### DISCUSSION TITLE
Why arent objects 'freed' when deleting content of a PVC? - Ceph RBD CSI

### DISCUSSION THREAD

**Discussion by mame-leadec** [ceph/ceph-csi#4618] [Q&A] [answered]
Hi everyone,

i am fairly new to Ceph / CSI drivers.

I have deployed a Ceph Cluster with RBD CSI driver via helm/rook,  created a PVC and a demo POD. 

It works as intended, but there is one thing I don't understand: 

- If i create a large file on the demo POD mounted storage, Ceph creates a lot of objects (Ok)
- When i delete this large file, the objects still remain (Why?)

(I'm just wondering, since the remaining objects from the already deleted large file are counted towards the total quota of available storage of the Ceph cluster)

1. Is this intented behaviour? Or do i have wrong understanding? 
2. Have I missed a configuration option to explicitly delete the objects?

Thanks.

**Comment by Madhu-1** [comment, marked as answer]
@mame-leadec that's because that is expected behaviour with RBD. we do have some automation for that to create the reclaimspace or you need to set the `discard` option in storageclass. https://github.com/rook/rook/issues/3832 and https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.5.0/docs/reclaimspace.md#reclaimspacejob

**Reply by mame-leadec** [reply]
Thanks for your fast answer and for pointing that out.

EDIT:
The `discard` option works. 

@Madhu-1 Just out of curiosity, what are the advantages to use the [reclaimspacejob](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.5.0/docs/reclaimspace.md#reclaimspacejob)?
